### DISCUSSION TITLE
Logout not working

### DISCUSSION THREAD

**Discussion by sureshadapa** [imixs/imixs-jsf-example#59] [Q&A]
LoginController and LogoutController missing in imixs-jsf-example

**Comment by rsoika** [comment]
You can take the class org.imixs.workflow.faces.util.LoginController - method doLogout()

http://www.imixs.org/doc/apidocs/org/imixs/workflow/faces/util/LoginController.html

The login and logout is completely transparent and part of the JAAS framework from Java Enterprise.
To logout you simply need to invalidate the session:
```
		FacesContext context = FacesContext.getCurrentInstance();
		ExternalContext externalContext = context.getExternalContext();
		HttpSession session = (HttpSession) externalContext.getSession(false);
		session.invalidate();
```
